app/controllers/profiles_controller.rb

```diff
@@ -67,29 +67,78 @@ def logged_in_admin
        redirect_to '/home'
      end
    end
-  
-  
+
+
   def show
     @profile = Profile.find(params[:id])
+    @image_url = profile_image(@profile, get_bucket)
   end
-  
+
   def display
     @allProfiles = Profile.all
   end
-  
+
   def destroy
     Profile.find(params[:id]).destroy
     flash[:success] = "Profile deleted"
 
     redirect_to '/search'
   end
-  
+
   def pictures
-    @allProfiles = Profile.all.order("last_name ASC, first_name ASC")
+    bucket = get_bucket
+
+    all_profiles = Profile.all.order("last_name ASC, first_name ASC")
+
+    @profiles = []
+    all_profiles.each do |profile|
+      @profiles.push({
+        :name => profile.last_name + ", " + profile.first_name,
+        :image_url => profile_image(profile, bucket),
+        :link => profile_path(profile.id)
+      })
+    end
   end
-  
+
   def get_dataset
     render json: { data: Profile.all }
   end
-end
 
+  private
+    def get_bucket
+      s3 = Aws::S3::Resource.new(
+        region: 'us-east-2',
+        credentials: Aws::Credentials.new(
+          ENV['AWS_ACCESS_KEY_ID'],
+          ENV['AWS_SECRET_ACCESS_KEY']
+        )
+      )
+
+      s3.bucket('mayflower-data')
+    end
+
+    def profile_image(profile, bucket)
+      oldname = profile.last_name.to_s + ", " + profile.first_name.to_s+".png"
+      oldname2 = profile.last_name.to_s + ", " + profile.first_name.to_s+".jpg"
```

Comment:
I understand `oldname` and `oldname2` are inherited from the previous code, but they should probably be changed to something more descriptive in this code refactor.